test/Microsoft.OpenApi.SmokeTests/ApiGurus.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Microsoft.OpenApi.Readers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.OpenApi.SmokeTests
+{
+    [Collection("DefaultSettings")]
+    public class ApisGuruTests
+    {
+        private static HttpClient _httpClient = new HttpClient(new HttpClientHandler() {
+            AutomaticDecompression = DecompressionMethods.GZip
+        });
+
+        private readonly ITestOutputHelper output;
+    
+        public ApisGuruTests(ITestOutputHelper output)
+        {
+           
+            this.output = output;
+        }
+
+        static ApisGuruTests()
+        {
+            _httpClient.DefaultRequestHeaders.AcceptEncoding.Add(new System.Net.Http.Headers.StringWithQualityHeaderValue("gzip"));
+            _httpClient.DefaultRequestHeaders.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("OpenApi.Net.Tests", "1.0"));
+           
+        }
+
+        public static IEnumerable<object[]> GetSchemas()
+        {
+
+            var listJsonStr = _httpClient
+                    .GetStringAsync("https://api.apis.guru/v2/list.json")
+                    .GetAwaiter().GetResult();
+
+                var json = JObject.Parse(listJsonStr);
+                foreach (var item in json.Properties())
+                {
+                    var versions = GetProp(item.Value, "versions") as JObject;
+                    if (versions == null)
+                        continue;
+                    foreach (var prop in versions.Properties())
+                    {
+                        var urlToJson = GetProp(prop.Value, "swaggerUrl")?.ToObject<string>();
+                        if (urlToJson != null)
+                            yield return new object[] { urlToJson };
+
+                        var utlToYaml = GetProp(prop.Value, "swaggerYamlUrl")?.ToObject<string>();
+                        if (utlToYaml != null)
+                            yield return new object[] { utlToYaml };
+                    }
+            }
+
+            JToken GetProp(JToken obj, string prop)
+            {
+                if (!(obj is JObject jObj))
+                    return null;
+                if (!jObj.TryGetValue(prop, out var jToken))
+                    return null;
+                return jToken;
+            }
+        }
+
+
+        [Theory(DisplayName = "APIs.guru")]
+        [MemberData(nameof(GetSchemas))]
+        public async Task EnsureThatICouldParse(string url)
+        {
+            var stopwatch = new Stopwatch();
+            
+            var response = await _httpClient.GetAsync(url);
+            if (!response.IsSuccessStatusCode)
+            {
+                output.WriteLine($"Couldn't load {url}");
```

Comment:
It's a test case. it can't understand to output the error message and returns directly.
Please use the assert clause to test the status code.

Reply:
These are smoke tests and shouldn't be treated the same way as integration tests.  If a URL cannot be resolved from apis.guru that is not a failure of our library, but an issue with an external directory, so I wouldn't want to flag it as an error.